src/EFCore.MySql/Migrations/MySqlMigrationsSqlGenerator.cs

```diff
@@ -758,7 +752,7 @@ protected virtual string[] GetColumns([NotNull] IEnumerable<IProperty> propertie
                 builder.Append(" NO MAXVALUE");
             }
 
-            builder.Append(operation.IsCyclic ? " CYCLE" : " NO CYCLE");
+            builder.Append(operation.IsCyclic ? " CYCLE" : " NOCYCLE");
```

Comment:
This should be implemented in an overridden [MigrationsSqlGenerator.SequenceOptions()](https://github.com/dotnet/efcore/blob/55e8f0a65e054ab6248474e4755d9e153c0f1c28/src/EFCore.Relational/Migrations/MigrationsSqlGenerator.cs#L1321) implementation.

Reply:
Am I wrong, or it seems that the only way to override this method is to rewrite it as we are doing? I don't see a way to change anything in MigrationCommandListBuilder builder after base method execution.

Reply:
Yes, you will basically have to copy the whole `SequenceOptions()` method, change `virtual` to `override` and then change `NO CYCLE` to `CYCLE`. I don't see any reliably shortcuts there.

Reply:
I updated the 2 Methods as we discussed,  Npgsql-like Implementation is better.
I've also implemented same modification on 5.0 branch in my repository, sorry, no time now to implement test cases so I didn't make the pull request. I think that by the end of week I will complete it

Reply:
Do you want that I PR it against 5.0 now, or after implementing test cases?